lib/ui/myplot/PlotList.dart

```diff
@@ -76,76 +75,103 @@ class _MyPlotState extends State<PlotList> {
-  footer: RoundedButton.build(style: LargeRoundedButtonStyle() ,context: context, title: viewModel.buttonTitle));
-}
 
-Widget _buildTitle(PlotListViewModel viewModel, PlotListStyle myPlotStyle, {BuildContext context}){
-  return Container(
-    padding: myPlotStyle.titleEdgePadding,
-    child: Row(
-      mainAxisAlignment: MainAxisAlignment.spaceBetween,
-        children: <Widget>[
-          Column(
-            mainAxisAlignment: MainAxisAlignment.start,
+  Widget _buildPage(BuildContext context, List<CropEntity> cropList,
+      PlotListStyle plotStyle, Function goToDetail) {
+    final viewModel = buildPlotListViewModel();
+    return HeaderAndFooterListView.builder(
+        itemCount: cropList.length,
+        itemBuilder: (BuildContext context, int index) {
+          final itemViewModel = fromCropEntityToViewModel(
+              cropList[index], goToDetail);
+          return PlotListItem().buildListItem(itemViewModel);
+        },
+        physics: ScrollPhysics(),
+        shrinkWrap: true,
+        header: _buildTitle(viewModel, plotStyle, context: context),
+        footer: RoundedButton(viewModel: RoundedButtonViewModel(title: viewModel.buttonTitle, onTap: () => _showToast(context)),
+            style: RoundedButtonStyle.largeRoundedButtonStyle()));
+  }
+
+  Widget _buildTitle(PlotListViewModel viewModel, PlotListStyle myPlotStyle,
+      {BuildContext context}) {
+    return Container(
+        padding: myPlotStyle.titleEdgePadding,
+        child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
             children: <Widget>[
-              Text(
-                viewModel.title,
-                style: myPlotStyle.titleTextStyle,
-              )
+              Column(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  children: <Widget>[
+                    Text(
+                      viewModel.title,
+                      style: myPlotStyle.titleTextStyle,
+                    )
+                  ]
+              ),
+              RoundedButton(viewModel:
+                  RoundedButtonViewModel(icon: "assets/icons/profit_add.png", onTap: () => _showToast(context)),
+                  style: RoundedButtonStyle.compactRoundedButton())
             ]
+        )
+    );
+  }
+
+  Widget _buildErrorPage(BuildContext context, MyPlotViewModel viewModel,
+      PlotListStyle plotStyle) {
+    final String retryButton = "Retry";
+    return Container(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: <Widget>[
+          AlertDialog(
+              title: Text(
+                  plotStyle.errorText),
+              actions: <Widget>[
+                FlatButton(
+                    child: Text(
+                        retryButton),
+                    onPressed: () {
+                      //FIXME: Needs to implement the retry functionality
```

Comment:
this is out of scope?

Reply:
This should call a ViewModel function. Suggest a var in the view model. ....  Function update;
The Bloc will then fill this in later.